File: Programming-Basics-With-JavaScript/Exams/01-Old-Exams/04-PB-Exam-July-2020/04.balls.js
function balls(input) {

    const ballsCount = Number(input[0]);
    let index = 1;

    let points = 0;
    let redBalls = 0;
    let orangeBalls = 0;
    let yellowBalls = 0;
    let whiteBalls = 0;
    let otherColorBalls = 0;
    let dividesCounter = 0;

    for (let i = 1; i <= ballsCount; i++) {
        // • Ако топката е “red” точките ни се повишават с 5.
        // • Ако топката е “orange” точките ни се повишават с 10.
        // • Ако топката е “yellow” точките ни се повишават с 15.
        // • Ако топката е “white” точките ни се повишават с 20.
        // • Ако топката е “black” точките ни се делят на 2, като закръгляме към по-малкото цяло число.
        let current = input[index++];
        switch (current) {
            case "red": points += 5;
                redBalls++;
                break;
            case "orange": points += 10;
                orangeBalls++;
                break;
            case "yellow": points += 15;
                yellowBalls++;
                break;
            case "white": points += 20;
                whiteBalls++;
                break;
            case "black": points = Math.floor(points / 2);
                dividesCounter++;
                break;
            default: otherColorBalls++;
                break;
        }
    }

    console.log(`Total points: ${points}`);
    console.log(`Red balls: ${redBalls}`);
    console.log(`Orange balls: ${orangeBalls}`);
    console.log(`Yellow balls: ${yellowBalls}`);
    console.log(`White balls: ${whiteBalls}`);
    console.log(`Other colors picked: ${otherColorBalls}`);
    console.log(`Divides from black balls: ${dividesCounter}`);
}
balls(["3"
    , "white"
    , "black"
    , "pink"]);

File: Python_Basics/Exams/07.exam_preparation_jun_2023/04.balls.py
balls_count = int(input())

points = 0
red_balls = 0
orange_balls = 0
yellow_balls = 0
white_balls = 0
black_balls = 0
other_color_balls = 0

for balls in range(balls_count):
    current_color = input()

    if current_color == 'red':
        # · Ако топката е "red" точките ни се повишават с 5.
        points += 5
        red_balls += 1
    elif current_color == 'orange':
        # · Ако топката е "orange" точките ни се повишават с 10.
        points += 10
        orange_balls += 1
    elif current_color == 'yellow':
        # · Ако топката е "yellow" точките ни се повишават с 15.
        points += 15
        yellow_balls += 1
    elif current_color == 'white':
        # · Ако топката е "white" точките ни се повишават с 20.
        points += 20
        white_balls += 1
    elif current_color == 'black':
        # · Ако топката е "black" точките ни се делят на 2 (целочислено).
        points //= 2
        black_balls += 1
    else:
        other_color_balls += 1

print(f'Total points: {points}')
print(f'Red balls: {red_balls}')
print(f'Orange balls: {orange_balls}')
print(f'Yellow balls: {yellow_balls}')
print(f'White balls: {white_balls}')
print(f'Other colors picked: {other_color_balls}')
print(f'Divides from black balls: {black_balls}')
